packages/amplify-category-api/src/provider-utils/awscloudformation/cdk-stack-builder/apigw-stack-transform.ts

```diff
@@ -186,23 +186,26 @@ export class ApigwStackTransform {
     const overrideFilePath = pathManager.getResourceDirectoryPath(undefined, AmplifyCategories.API, this.resourceName);
     const overrideJSFilePath = path.join(overrideFilePath, 'build', 'override.js');
 
-    const isBuild = await buildOverrideDir(backendDir, overrideFilePath).catch(error => {
-      printer.debug(`Build error : ${error.message}`);
-      return false;
-    });
+    const isBuild = await buildOverrideDir(backendDir, overrideFilePath);
 
     // skip if packageManager or override.ts not found
     if (isBuild) {
-      const { override } = await import(overrideJSFilePath).catch(error => {
+      let override;
+      try {
+        ({ override } = await import(overrideJSFilePath));
```

Comment:
nit: does this statement require outer brackets ()?

Reply:
Parenthesis are required when assignment is separate from declaration: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#object_destructuring